python_modules/dagster/dagster/core/execution_plan/simple_engine.py

```diff
@@ -184,7 +179,17 @@ def execute_step_in_memory(step_context, inputs, intermediates_manager):
         yield DagsterEvent.step_failure_event(
             step_context=step_context, step_failure_data=StepFailureData(error=error_info)
         )
+
+        if step_context.executor_config.throw_on_user_error:
+            raise dagster_error
```

Comment:
should this be `if throw_on_user_error and dagster_error.is_user_code_error`?

Reply:
I think the answer is "no" and it is "no" because "throw_on_user_error" is the wrong thing. What do you think the behavior should be on a framework error?

Reply:
Ya its still not clear to me what exactly the behavior should be which is i guess the bigger question i should have asked. 

One path is:
* throwing - Any exception that occurs that would cause pipeline failure is rethrown. User code exceptions are wrapped. 
* nonthrowing - Exceptions are reported but not thrown. Any time the process ends in an unhandled exception is a bug.